src/azure-cli/azure/cli/command_modules/ams/commands.py

```diff
@@ -66,7 +66,8 @@ def get_custom_sdk(custom_module, client_factory):
                          custom_command_type=get_custom_sdk('mru', None))
 
     with self.command_group('ams transform', get_sdk('Transforms', get_transforms_client)) as g:
-        g.show_command('show', 'get')
+        g.custom_command('show', 'get_transform',
```

Comment:
If you just want to make show command return 3, you just need to add an exception handler with https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/arm.py#L754.
e.g. https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/storage/commands.py#L379

Reply:
We cannot use the exception handler in arm.py. When we first published our API, we decided on a programming model that would allow for the following (not python)
o = get(transform)
if o == null
instead of:
try
{
    o = get(transform)
}
catch()
{}
In our swaggers, we explicitly define a 404 response, so we don't throw an exception and instead just return null.